Test Failure (20180420-0319): JDBC41UpgradeTest.testJDBC41Tolerance40Driver AccessControlException: Access denied MBeanServerPermission

Fails with Java 2 security enabled:

Test Failure (20180420-0319): com.ibm.ws.jdbc.fat.v41.JDBC41UpgradeTest.testJDBC41Tolerance40Driver

junit.framework.AssertionFailedError: 2018-04-20-07:35:53:115 Expected response 200, received 500 (Internal Server Error) while connecting to http://localhost:8010/basicfat/BasicTestServlet?testMethod=testJDBC41Tolerance40Driver. Error stream: Error 500: java.lang.Exception: java.lang.ExceptionInInitializerError

	at componenttest.topology.utils.HttpUtils.getHttpConnection(HttpUtils.java:289)
	at componenttest.topology.utils.HttpUtils.getHttpConnection(HttpUtils.java:215)
	at componenttest.topology.utils.HttpUtils.findStringInUrl(HttpUtils.java:471)
	at componenttest.topology.utils.HttpUtils.findStringInReadyUrl(HttpUtils.java:444)
	at componenttest.topology.utils.HttpUtils.findStringInReadyUrl(HttpUtils.java:414)
	at componenttest.topology.utils.FATServletClient.runTest(FATServletClient.java:81)
	at componenttest.topology.utils.FATServletClient.runTest(FATServletClient.java:70)
	at com.ibm.ws.jdbc.fat.v41.JDBC41UpgradeTest.testJDBC41Tolerance40Driver(JDBC41UpgradeTest.java:48)
	at componenttest.custom.junit.runner.FATRunner$1.evaluate(FATRunner.java:193)
	at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:315)
	at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:167)


Full stack:
Stack Dump = java.lang.Exception: java.lang.ExceptionInInitializerError
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.loadServlet(ServletWrapper.java:1630)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:587)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:440)
	at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1154)
	at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest(WebApp.java:4962)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest(DynamicVirtualHost.java:314)
	at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:996)
	at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:279)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1011)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:414)
	at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:373)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:530)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:464)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:329)
	at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:300)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:165)
	at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:74)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:501)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:571)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:926)
	at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1015)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1153)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.lang.Thread.run(Thread.java:785)
Caused by: java.lang.ExceptionInInitializerError
	at java.lang.J9VMInternals.ensureError(J9VMInternals.java:141)
	at java.lang.J9VMInternals.recordInitializationFailure(J9VMInternals.java:130)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:88)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:57)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:437)
	at com.ibm.ws.managedobject.internal.ManagedObjectFactoryImpl.createManagedObject(ManagedObjectFactoryImpl.java:64)
	at com.ibm.ws.webcontainer.osgi.webapp.WebApp.inject(WebApp.java:1264)
	at com.ibm.ws.webcontainer.osgi.webapp.WebApp.injectAndPostConstruct(WebApp.java:1440)
	at com.ibm.ws.webcontainer.osgi.webapp.WebApp.injectAndPostConstruct(WebApp.java:1428)
	at com.ibm.ws.webcontainer.osgi.servlet.ServletWrapper.createTarget(ServletWrapper.java:63)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper$1.run(ServletWrapper.java:1535)
	at java.security.AccessController.doPrivileged(AccessController.java:650)
	at com.ibm.ws.webcontainer.servlet.ServletWrapper.loadServlet(ServletWrapper.java:1503)
	... 23 more
Caused by: java.security.AccessControlException: Access denied ("javax.management.MBeanServerPermission" "createMBeanServer")
	at java.security.AccessController.throwACE(AccessController.java:157)
	at java.security.AccessController.checkPermissionHelper(AccessController.java:217)
	at java.security.AccessController.checkPermission(AccessController.java:349)
	at java.lang.SecurityManager.checkPermission(SecurityManager.java:562)
	at java.lang.management.ManagementFactory.getPlatformMBeanServer(ManagementFactory.java:368)
	at jdbc.fat.v41.web.BasicTestServlet.<clinit>(BasicTestServlet.java:73)
	... 35 more

--- a/dev/com.ibm.ws.jdbc_fat_v41/publish/servers/com.ibm.ws.jdbc.fat.v41/server.xml
+++ b/dev/com.ibm.ws.jdbc_fat_v41/publish/servers/com.ibm.ws.jdbc.fat.v41/server.xml
@@ -62,5 +62,9 @@
     
     <javaPermission codebase="${server.config.dir}/derby/derby.jar" className="java.security.AllPermission"/>
 
+    <!--  Permissions for application to access mbeans -->
+    <javaPermission codebase="${server.config.dir}/apps/basicfat.war" className="javax.management.MBeanPermission" actions="queryNames"/>
+    <javaPermission codebase="${server.config.dir}/apps/basicfat.war" className="javax.management.MBeanServerPermission" name="createMBeanServer"/>
+
     <variable name="onError" value="FAIL"/>
 </server>

--- a/dev/com.ibm.ws.jdbc_fat_v41/test-applications/basicfat/src/jdbc/fat/v41/web/BasicTestServlet.java
+++ b/dev/com.ibm.ws.jdbc_fat_v41/test-applications/basicfat/src/jdbc/fat/v41/web/BasicTestServlet.java
@@ -70,7 +70,6 @@ public class BasicTestServlet extends FATDatabaseServlet {
     private List<String> globalSchemaList = null;
     private boolean isGetColorRegistered = false;
     private boolean isGetUserRegistered = false;
-    public static MBeanServer mbeanServer = ManagementFactory.getPlatformMBeanServer();
 
     @Resource
     private UserTransaction tran;

